action.js

```diff
@@ -0,0 +1,192 @@
+//
+// This agent will read the user input and the GitHub context object to
+// determine what metrics should be stored, and what values each stored metric
+// should use for the various parameters.
+//
+//
+
+const cloneDeep = require('lodash.clonedeep');
+const has = require('lodash.has');
+
+// Determine if GitHub context is a pull request
+const isPullRequest(github) {
+  return has(github, ["context","eventName"]) && github.context.eventName === "pull_request");
+}
+
+//
+//  PURPOSE
+//
+// This function generates the best-effort payload for the Events API based
+// entirely on the github context object. This is meant to be used when the
+// user has not supplied any information.
+//
+// As of now, Pull Requests events are caught and translated into change
+// events, when applicable. Other events are also stored, but are not
+// transformed much.
+//
+//
+//  INPUT
+//
+// team_id  string  supplied by CTO.ai to a member of the team.
+// github   object  large object of metadata from GitHub Actions.
+//
+//
+//  OUTPUT
+//
+// The function outputs a JavaScript object to use for the body of the HTTP API
+// request, passed to the sendEvent function.
+//
+//
+//  EXAMPLE
+//
+// const http_body = extractBody("aef3d53", {context:{}});
+//
+//
+const extractBody = (team_id, github) => {
+
+  //
+  // TODO1: Capture deployment events (see "GitHub deployment events")
+  // TODO2: Capture other events in the Lifecycle Catalog
+  //
+  //
+
+  // Detect Change Initiated (Pull Request Opened against Master)
+  if (
+
+// is a pull request
+has(github, ["context","eventName"]) && github.context.eventName === "pull_request" &&
+
+// is against master
+has(github, ["context","payload","pull_request","base","ref"]) && github.context.payload.pull_request.base.ref === "master" &&
+
+// is opened
+has(github, ["context","payload","action"]) && github.context.payload.action === "opened"
+
+     ) {
+    return ({
+      stage: "Change",
+      status: "Initiated",
+      change_id: github.context.payload.pull_request.head.ref,
+      team_id,
+      custom: github
+    });
+  }
+
+  // Detect Change Succeeded (Pull Request Closed against Master)
+  if (has(github, ["context","eventName"]) &&
+      github.context.eventName === "pull_request" &&
```

Comment:
Can we do `github?.context?.eventName === "pull_request" instead of using lodash?